java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -617,4 +620,122 @@ public void exists_multiple_keys(BaseClient client) {
                 client.exists(new String[] {key1, key2, key1, UUID.randomUUID().toString()}).get();
         assertEquals(3L, existsKeysNum);
     }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void expire_pexpire_and_ttl_with_positive_timeout(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        assertEquals(OK, client.set(key, "expire_timeout").get());
+        assertTrue(client.expire(key, 10L).get());
+        assertTrue(client.ttl(key).get() <= 10L);
+
+        // set command clears the timeout.
+        assertEquals(OK, client.set(key, "pexpire_timeout").get());
+        if (REDIS_VERSION.feature() < 7) {
+            assertTrue(client.pexpire(key, 10000L).get());
+        } else {
+            assertTrue(client.pexpire(key, 10000L, ExpireOptions.HAS_NO_EXPIRY).get());
+        }
+        assertTrue(client.ttl(key).get() <= 10L);
+
+        // TTL will be updated to the new value = 15
+        if (REDIS_VERSION.feature() < 7) {
+            assertTrue(client.expire(key, 15L).get());
+        } else {
+            assertTrue(client.expire(key, 15L, ExpireOptions.HAS_EXISTING_EXPIRY).get());
+        }
+        assertTrue(client.ttl(key).get() <= 15L);
+    }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void expireAt_pexpireAt_and_ttl_with_positive_timeout(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        assertEquals(OK, client.set(key, "expireAt_timeout").get());
+        assertTrue(client.expireAt(key, Instant.now().getEpochSecond() + 10L).get());
+        assertTrue(client.ttl(key).get() <= 10L);
+
+        // extend TTL
+        if (REDIS_VERSION.feature() < 7) {
+            assertTrue(client.expireAt(key, Instant.now().getEpochSecond() + 50L).get());
+        } else {
+            assertTrue(
+                    client
+                            .expireAt(
+                                    key,
+                                    Instant.now().getEpochSecond() + 50L,
+                                    ExpireOptions.NEW_EXPIRY_GREATER_THAN_CURRENT)
+                            .get());
+        }
+        assertTrue(client.ttl(key).get() <= 50L);
+
+        if (REDIS_VERSION.feature() >= 7) {
+            // set command clears the timeout.
+            assertEquals(OK, client.set(key, "pexpireAt_timeout").get());
+            assertFalse(
+                    client
+                            .pexpireAt(
+                                    key, Instant.now().getEpochSecond() + 50000L, ExpireOptions.HAS_EXISTING_EXPIRY)
```

Comment:
is this how we send time in milliseconds? in python, we need to multiply by 1000

Reply:
lets add an else and test with no expiry options

Reply:
No you are right I need to multiply by a 1000 :D Will fix them.